servers/src/main/java/tachyon/master/next/filesystem/meta/InodeTree.java

```diff
@@ -220,7 +220,7 @@ public void deleteInode(Inode inode) throws FileDoesNotExistException {
 
     mInodes.remove(inode);
     mPinnedInodeFileIds.remove(inode.getId());
-    inode.reverseId();
+    inode.delete();
```

Comment:
I am not sure if there' code actually checks for "id < 0". I hope not, because that means we expose this internal logic to external world, and it'll be very difficult to audit - someone has to track the data flow of the id, e.g. via assignment, and in some cases it's nearly impossible. That's why I advocate using a flag. 

Put that aside, I'm not sure if we want to remove it and put a TODO, because we have delete() here in inodeTree and I feel its safer to mark inode as deleted in addition to removing it from Inodes. Thoughts?